src/kbucket.rs

```diff
@@ -1,146 +1,148 @@
+            FindNodeResult::NotFound(known_nodes)
         }
     }
+    // TODO:
+    pub fn find_value() {}
 
     // TODO:
-    fn add_store(&self) {
-    
-    }
+    /// Instructs a node to store a key, value pair for later retrieval. "Most operations are implemented
+    /// in terms of the lookup proceedure. To store a <key,value> pair, a participant locates the k closes
+    /// nodes to the key and sends them store RPCs".
+    pub fn store(&mut self, key: Identifier, value: Vec<u8>) {}
 
-    fn find_bucket(&self, identifier: Identifier) -> usize {
-        let x = U256::from(self.local_node_id);
-        let y = U256::from(identifier);
-        let xor_distance = x^y;
-        
-        let bucket_index = ((256 - xor_distance.leading_zeros()) as usize);
-        println!("Xor distance leading zeros, {}", xor_distance.leading_zeros());
-        println!("Bucket index for given key: {}", bucket_index);
-        bucket_index
+    // Non-RPCs:
+    // ---------------------------------------------------------------------------------------------------
+    pub fn add_node(&mut self, node: Node) {
+        let result = self.search_table(node.node_id);
+        let mut bucket = self.buckets[result.bucket_index];
+
+        if !result.found {
+            bucket[result.column_index] = Some(node)
+        } else {
+            println!("Node is already in our table")
+        }
     }
 
-    // How can i make this return value less confusing?
-    fn search_bucket(&self, bucket: Bucket, node: Node) -> (bool, usize) {
+    // Searches table for node specified.
+    fn search_table(&self, id: Identifier) -> SearchResult {
         let mut last_empty_index = 0;
-        for i in 0..BUCKET_SIZE { 
-            match bucket[i] {
+        let bucket_index = self.find_bucket_index(id);
+        let mut bucket = self.buckets[bucket_index];
+
+        for (i, node) in bucket.iter().enumerate() {
+            match node {
                 Some(bucket_node) => {
-                    // If node was already in bucket -->  return (it's index, true).
-                    if bucket_node == node {
-                        return (true, i)
-                    }
-                    else {continue};
+                    if bucket_node.node_id == id {
+                        SearchResult {
+                            found: true,
+                            bucket_index,
+                            column_index: i,
+                        }
+                    } else {
+                        continue;
+                    };
                 }
-                None => {
+                _ => {
                     last_empty_index = i;
                 }
             }
         }
-        // If node wasn't already in bucket -->  return (largest available index, false)
-        println!("Last empty index: {}", last_empty_index);
-        return (false, last_empty_index)
+        SearchResult {
+            found: false,
```

Comment:
if found is helpful info beyond just the binary "was or was not here" then instead I'd suggest an `enum` as something a bit more "idiomatic"

```rust
enum Search {
    Success(usize,usize),
    Failure(usize,usize),
}
```

Reply:
This seems like the move.  Going to modify code in test branch (sorry) to use this.

Reply:
Actually might hold off on this... Everything breaks when I move from the SearchResult struct.  I don't think it's worth the headache.  Thoughts?

Reply:
Implemented here:  https://github.com/EchoAlice/my_kademlia/commit/34e5d5b41a49e860e4e6e54facdafef366381c1e